### PR TITLE
[ember-data] Export members of `DS` namespace

### PR DESCRIPTION
**Pull request by charlesfries**
Starting on some of the changes requested in https://github.com/typed-ember/ember-cli-typescript/issues/1431

One concern: is exporting members explicitly by name (currently only `Snapshot` in `-private.d.ts`) the only way to get the types out of the `DS` namespace? Is there a spread-like syntax that will just export `DS.*` as individual named exports?

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

[Documentation URL](https://github.com/typed-ember/ember-cli-typescript/issues/1431)

### CODE REVIEW

**Comment by chriskrycho** [comment]
@charlesfries sorry about the delay—I will try to take a look at this (hopefully with some input from @runspired) in the next couple days.

**Review by chriskrycho — CHANGES_REQUESTED** [review]

File: types/ember-data__store/-private.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import DS from 'ember-data';
+
+export import Snapshot = DS.Snapshot;
```
[the remark this answers is not in the document]
Comment (runspired) [reply types/ember-data__store/-private.d.ts:3]: Sorry, I wrote this like a week ago but somehow it turned out to have not published

[the remark this answers is not in the document]
Comment (charlesfries) [reply types/ember-data__store/-private.d.ts]: With that change I hit these errors:

`Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.`

`A 'const' initializer in an ambient context must be a string or numeric literal or literal enum reference.`

Adding `declare`  to the const fixes the first issue but I'm not sure about the second. I resubmitted a version that just exports the const directly.

[the remark this answers is not in the document]
Comment (charlesfries) [reply types/ember-data__store/-private.d.ts]: > If the goal is simply a place to import this from that isn't the deprecated DS namespace

Yes, that was my initial (personal) motivation for this PR.

I went ahead and moved the export to `Store`. Lmk if you'd rather I create that new package.

**Comment by RyanCavanaugh** [comment]
This seems fine to me but I'd really like one of the owners to ✔...

**Comment by chriskrycho** [comment]
Ugh, sorry. Fell off my radar and the other two maintainers have had COVID. I’ll look at it again on Monday.

File: types/ember-data__store/-private.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (chriskrycho) [reply types/ember-data__store/-private.d.ts]: Moving the export to `Store` and not *yet* creating the new package seems better. 👍🏼

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
@charlesfrieds if you have bandwidth to help push on the TS story for Ember Data, I am going to have a task written up at the end of the week to get us to a place where we can publish preview types from Ember Data directly (very similarly to [what we’ve done for Ember itself][ember-preview]), which will in turn let us do things like introduce and restructure packages along the lines @runspired suggested much more easily.

[ember-preview]: https://blog.emberjs.com/announcing-official-typescript-types-public-preview

**Comment by chriskrycho** [comment]
@charlesfries if you have bandwidth to help push on the TS story for Ember Data, I am going to have a task written up at the end of the week to get us to a place where we can publish preview types from Ember Data directly (very similarly to [what we’ve done for Ember itself][ember-preview]), which will in turn let us do things like introduce and restructure packages along the lines @runspired suggested much more easily.

[ember-preview]: https://blog.emberjs.com/announcing-official-typescript-types-public-preview

**Comment by RyanCavanaugh** [comment]
Much appreciated @chriskrycho

**Comment by chriskrycho** [comment]
No problem, and thanks for the nudge!

**Pull request merged**